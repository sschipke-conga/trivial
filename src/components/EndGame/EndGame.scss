.end-game-section {
  background: url(../../images/confetti.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: 2em;
  color: white;
  height: 100vh;
}

.end-game-div {
  background: rgba(0, 0, 0, .9);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

}