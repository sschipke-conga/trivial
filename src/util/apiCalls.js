export const fetchQuestions = async (amount, category, difficulty) => {
  let url = `https://opentdb.com/api.php?amount=${amount}&category=${category}&difficulty=${difficulty}&type=multiple`
  const res = await fetch(url);
  if (!res.ok) {
    throw Error('Failed to fetch')
  }
  const data = await res.json()
  return data.results
}
