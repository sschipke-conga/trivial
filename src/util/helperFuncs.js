export const shuffle = answers => {
  var m = answers.length, t, i;
  while (m) {
    i = Math.floor(Math.random() * m--);
    t = answers[m];
    answers[m] = answers[i];
    answers[i] = t;
  }
  return answers;
}

export const determineCurrentTeam = (turnCount) => {
  if (turnCount % 2 === 0) {
    return 1
  } 
  return 0
}

export const determineWinner = teams => {
  if(teams[0].score === teams[1].score) {
    return false
  }
  return teams.sort((teamOne, teamTwo) => {
    return teamTwo.score-teamOne.score
  })[0]
}

export const capDifficulty = difficulty => {
  return difficulty[0].toUpperCase() + difficulty.slice(1)
}

export const cleanEncoding = (string) => {
  let cleanPhrase = string.replace(/&#039;/g, "'")
  .replace(/&quot;/g, '"')
  .replace(/&eacute/g, 'Ã©');
  return cleanPhrase
}

export const cleanQuestions = (questions) => {
    return questions.map(question => {
    question.question = cleanEncoding(question.question)
    question.correct_answer = cleanEncoding(question.correct_answer)
    question.incorrect_answers = question.incorrect_answers.map(answer => cleanEncoding(answer))
    return question
    })
}