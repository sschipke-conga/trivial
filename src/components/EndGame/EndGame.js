import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {determineWinner} from '../../util/helperFuncs';
import './EndGame.scss'

export const EndGame = ({teams, turnCount}) => {
  const winningTeam = determineWinner(teams)

  return(
    <section className="end-game-section">
      <h1 className="congrats">Congratulations!</h1>
    { winningTeam && <div className="ende-game-div">
    <h2 className="winning-team">{winningTeam.name} you win!</h2>
    <p className="winning-questions">You answered {winningTeam.score} questions correctly!</p>
    </div>}
    {!winningTeam && <h2>It's a tie! You both win!</h2>}

    </section>
  )
}

export const mapStateToProps = state => ({
  teams: state.teams,
  turnCount: state.turnCount
})

export default connect(mapStateToProps)(EndGame);

EndGame.propTypes = {
  teams: PropTypes.array.isRequired,
  turnCount: PropTypes.number
}