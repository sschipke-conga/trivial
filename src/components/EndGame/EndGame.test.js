import React from 'react';
import { shallow } from 'enzyme';
import {determineWinner} from '../../util/helperFuncs';
import {EndGame, mapStateToProps } from './EndGame';

jest.mock('../../util/helperFuncs')

const mockTeams = [{
  name: 'Lez Quizzerable',
  score: 2
}, {
  name: "Let's Get Quizzacle",
  score: 0
}]

describe('EndGame', () => {
  determineWinner.mockImplementation(() => mockTeams[0])
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<EndGame 
    teams={mockTeams}
    />)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should call determineWinner', () => {
    expect(determineWinner).toHaveBeenCalledWith(mockTeams)
  })
  it('mapStateToProps gives all the teams to state', () => {
    const mockState = {
      teams: mockTeams
    };
    const expected = {
      teams: mockState.teams
    };
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
  })
})

describe('EndGame with ties', () => {
  const mockTieTeams = [{
    name: 'Lez Quizzerable',
    score: 2
  }, {
    name: "Let's Get Quizzacle",
    score: 2
  }]
  let wrapper;
  determineWinner.mockImplementation(() => false)
  beforeEach(() => {
    wrapper = shallow(<EndGame
      teams={mockTieTeams}
    />)
  })
  it('should match the snapshot if there is no winner (a tie)', () => {
    expect(wrapper).toMatchSnapshot()
  })
})