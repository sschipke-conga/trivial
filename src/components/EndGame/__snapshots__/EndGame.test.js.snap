// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndGame should match the snapshot 1`] = `
<section
  className="end-game-section"
>
  <div
    className="end-game-div"
  >
    <h1
      className="congrats"
    >
      Congratulations!
    </h1>
    <h2>
      It's a tie! You both win!
    </h2>
  </div>
</section>
`;

exports[`EndGame with ties should match the snapshot if there is no winner (a tie) 1`] = `
<section
  className="end-game-section"
>
  <div
    className="end-game-div"
  >
    <h1
      className="congrats"
    >
      Congratulations!
    </h1>
    <h2>
      It's a tie! You both win!
    </h2>
  </div>
</section>
`;
